Sources/XcodeProjMapper/Mappers/Targets/PBXTarget+PlatformInference.swift

```
@@ -0,0 +1,85 @@
+    ///
+    /// - Returns: A `Destinations` set representing all supported destinations.
+    /// - Throws: If retrieving deployment targets fails.
+    func platform() throws -> Destinations {
+        if let sdkName = buildConfigurationList?.stringSetting(for: .sdkroot),
+           let root = Platform(sdkroot: sdkName)
+        {
+            return root.destinations
+        } else {
+            return try inferPlatformFromTarget()
+        }
+    }
+
+    /// Infers the platform from deployment targets if `SDKROOT` is not set or recognized.
+    ///
+    /// Aggregates all platforms indicated by the deployment targets. If none are found,
+    /// picks a default based on product type:
+    /// - iOS for most apps, clips, and app extensions.
+    /// - macOS for frameworks, libraries, command line tools, macros, xpc, system extensions.
+    /// - tvOS for tvTopShelfExtension.
+    /// - iOS otherwise.
+    ///
+    /// - Returns: A `Destinations` set representing all inferred destinations.
+    /// - Throws: If retrieving deployment targets fails.
+    private func inferPlatformFromTarget() throws -> Destinations {
+        let deploymentTargets = try deploymentTargets()
+        var result = Destinations()
+
+        if deploymentTargets.iOS != nil {
+            result.formUnion(Platform.iOS.destinations)
+        }
+        if deploymentTargets.macOS != nil {
+            result.formUnion(Platform.macOS.destinations)
+        }
+        if deploymentTargets.watchOS != nil {
+            result.formUnion(Platform.watchOS.destinations)
+        }
+        if deploymentTargets.tvOS != nil {
+            result.formUnion(Platform.tvOS.destinations)
+        }
+        if deploymentTargets.visionOS != nil {
+            result.formUnion(Platform.visionOS.destinations)
+        }
+
+        guard result.isEmpty else { return result }
+
+        // Fallback if no platform detected.
+        let product = productType?.mapProductType() ?? .app
```

Should we consider throwing instead of falling back to `.app`?